karbon/k8s_registration_structs.go

```diff
@@ -0,0 +1,102 @@
+package karbon
```

Comment:
Where are these structs coming from? Are they generated from a OpenAPI Spec? If so, we should ideally also check that spec in and do the generation via the makefile.

Reply:
The structure is generated from https://github.com/nutanix-core/acs-k8s/blob/master/api/modules/v1/cluster_registration_defs.yaml. Could you please tell me if it is ok for us to include the yaml in this repository for confidentiality purpose? thanks

Reply:
I have checked the structure here https://github.com/nutanix-cloud-native/prism-go-client/blob/main/karbon/karbon_cluster_structs.go and seems the structure is also define in the go file.

Reply:
That's a good question. @deepakm-ntnx thoughts?

Reply:
ideally we should generate them and store them per version folder (v1, v1alpha1) so that evenif it changes, we can maintain different versions.

Reply:
After tried for the autogen using the swagger for the structure, I have done some investigation is that the swagger is a api generator and since the api is not in the prism go client. If we just want to generate the struct it has to be implemented. 
```
GO111MODULE=on go build -mod vendor -o bin/nutanixclient .
swagger validate api/modules/v1/cluster_registration_defs.yaml
2023/01/05 22:33:58 
The swagger spec at "api/modules/v1/cluster_registration_defs.yaml" showed up some valid but possibly unwanted constructs.
2023/01/05 22:33:58 See warnings below:
2023/01/05 22:33:58 - WARNING: definition "#/definitions/k8s_create_cluster_registration_request" is not used anywhere
2023/01/05 22:33:58 - WARNING: definition "#/definitions/k8s_create_cluster_registration_response" is not used anywhere
2023/01/05 22:33:58 - WARNING: definition "#/definitions/k8s_cluster_registration_list" is not used anywhere
2023/01/05 22:33:58 - WARNING: definition "#/definitions/k8s_cluster_registration_delete_response" is not used anywhere
```
@deepakm-ntnx @thunderboltsid Do you think we could keep the struct to be manually defined under the prism-go-client/karbon? 
In the acs-k8s we also implemented the apis so thats the reason swagger could be used